src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala

```diff
@@ -0,0 +1,1433 @@
+          image = "private/image",
+          pullConfig = Option(DockerPullConfig("pullConfigSecret")))))
+      val app = App(
+        id = "/app", cmd = Some("cmd"), container = Option(container))
+      val (body, plan) = prepareApp(app, groupManager)
+
+      When("The create request is made")
+      clock += 5.seconds
+      val entity = HttpEntity(body).withContentType(ContentTypes.`application/json`)
+      Post(Uri./, entity) ~> route ~> check {
+        Then("It fails")
+        status shouldEqual StatusCodes.UnprocessableEntity withClue s"body: ${ByteString(body).utf8String}, response: ${responseAs[String]}"
+        responseAs[String] should include("pullConfig.secret must refer to an existing secret")
+      }
+    }
+
+    "Creation of an app with valid pull config should fail if secrets feature is disabled" in new Fixture {
+      Given("An app with a Docker config.json")
+      val container = RamlContainer(
+        `type` = EngineType.Mesos,
+        docker = Option(DockerContainer(
+          image = "private/image",
+          pullConfig = Option(DockerPullConfig("pullConfigSecret")))))
+      val app = App(id = "/app", cmd = Some("cmd"), container = Option(container))
+      val (body, plan) = prepareApp(app, groupManager)
+
+      When("The create request is made")
+      clock += 5.seconds
+      val entity = HttpEntity(body).withContentType(ContentTypes.`application/json`)
+      Post(Uri./, entity) ~> route ~> check {
+        Then("It fails")
+        status shouldEqual StatusCodes.UnprocessableEntity withClue s"body: ${ByteString(body).utf8String}, response: ${responseAs[String]}"
+
+        val responseStr = responseAs[String]
+        responseStr should include("/container/docker/pullConfig")
+        responseStr should include("must be empty")
+        responseStr should include("Feature secrets is not enabled. Enable with --enable_features secrets)")
+      }
+    }
+
+    "Fail creating application when network name is missing" in new Fixture {
+      Given("An app and group")
+      val app = App(
+        id = "/app",
+        cmd = Some("cmd"),
+        networks = Seq(Network(mode = NetworkMode.Container))
+      )
+
+      When("The create request is made")
+      val entity = HttpEntity(Json.stringify(Json.toJson(app)).getBytes("UTF-8")).withContentType(ContentTypes.`application/json`)
+      Post(Uri./, entity) ~> route ~> check {
+        Then("Validation fails")
+        status shouldEqual StatusCodes.UnprocessableEntity
+        responseAs[String] should include(NetworkValidationMessages.NetworkNameMustBeSpecified)
+      }
+    }
+
+    "Create a new app with IP/CT, no default network name, Alice does not specify a network" in new Fixture {
+      Given("An app and group")
+      val app = App(
+        id = "/app",
+        cmd = Some("cmd"),
+        networks = Seq(Network(mode = NetworkMode.Container))
+      )
+      the[NormalizationException] thrownBy {
+        prepareApp(app, groupManager)
```

Comment:
Is this the right spot for testing this? Isn't this independent of the controller?